scripts/documentation/plugin-quality-indicators.ts

```diff
@@ -0,0 +1,160 @@
+      npmData.githubDirectory
+    );
+    qualityIndicators[plugin.name] = {
+      lastPublishedDate: npmData.lastPublishedDate,
+      githubDirectory: npmData.githubDirectory,
+      npmDownloads,
+      githubStars,
+      nxVersion,
+    };
+    console.log(qualityIndicators[plugin.name]);
+  }
+
+  writeFileSync(
+    '../../nx-dev/nx-dev/pages/extending-nx/quality-indicators.json',
+    JSON.stringify(qualityIndicators, null, 2)
+  );
+}
+main();
+
+// Publish date (and github directory, readme content)
+// i.e. https://registry.npmjs.org/@nxkit/playwright
+async function getNpmData(plugin: PluginRegistry) {
+  const { data } = await axios.get(`https://registry.npmjs.org/${plugin.name}`);
+  console.log(plugin.name);
+  const lastPublishedDate = data.time[data['dist-tags'].latest];
+  if (!data.repository) {
+    console.log('!!! No repository!');
+    return { lastPublishedDate, githubRepo: '' };
+  }
+  const url: String = data.repository.url;
+  const githubRepo = url
+    .slice(url.indexOf('github.com/') + 11)
+    .replace('.git', '');
+  return {
+    lastPublishedDate,
+    githubDirectory: data.repository.directory,
+    githubRepo,
+    // readmeContent: plugin.name
+  };
+}
+
+// Download count
+// i.e. https://api.npmjs.org/downloads/point/2023-06-01:2023-07-01/@nxkit/playwright
+async function getNpmDownloads(plugin: PluginRegistry) {
+  const lastMonth = getLastMonth();
+  const { data } = await axios.get(
+    `https://api.npmjs.org/downloads/point/${stringifyIntervalForUrl(
+      lastMonth
+    )}/${plugin.name}`
+  );
+  return data.downloads;
+}
+
+export function getLastMonth() {
+  const now = new Date();
+  const thisYear = now.getFullYear();
+  const thisMonth = now.getMonth();
+  return {
+    start: new Date(`${thisYear}-${thisMonth - 1}-01`),
+    end: new Date(`${thisYear}-${thisMonth}-01`),
+  };
+}
+
+export function stringifyIntervalForUrl(interval: Interval): string {
+  return `${stringifyDate(interval.start)}:${stringifyDate(interval.end)}`;
+}
+
+export function stringifyDate(date: Date) {
+  // yyyy-MM-dd
+  return date.toISOString().slice(0, 10);
+}
+
+// Stars
+// i.e. https://api.github.com/repos/nxkit/nxkit
+async function getGithubStars(repo: String) {
+  try {
+    const { data } = await axios.get(`https://api.github.com/repos/${repo}`, {
+      headers: {
+        Authorization: `Bearer ${process.env.GITHUB_PAT}`,
+      },
+    });
+    return data.stargazers_count;
+  } catch (ex) {
+    return -1;
+  }
+}
+
+// Nx Version
+// i.e. https://raw.githubusercontent.com/nxkit/nxkit/HEAD/packages/playwright/package.json
+// i.e. https://raw.githubusercontent.com/nxkit/nxkit/HEAD/package.json
+async function getNxVersion(githubRepo: String, githubDirectory?: String) {
+  if (!githubRepo) {
+    return '';
+  }
+  if (githubDirectory) {
+    try {
+      const { data } = await axios.get(
+        `https://raw.githubusercontent.com/${githubRepo}/HEAD/${githubDirectory}/package.json`
+      );
```

Comment:
This should come from NPM right?

And actually, could we have a range? The range should be the `peerDependencies` of the `@nx/devkit` version that the plugin depends on.

        "nx": ">= 14.1 <= 16"


        "nx": ">= 15 <= 17"


I think the logic would be

1. Get `@nx/devkit` version in the `dependencies` or `peerDependencies` (it should really be in `dependencies` though.)
2. Get `nx` version in `peerDependencies` the version of `@nx/devkit`) above.


And for filtering, this is a semver range, so we should be able to use semver or something else to see if a certain version of Nx is supported or not.